### ISSUE TITLE
"Opt-out of search engine indexing" considered harmful

### ISSUE THREAD

**Issue by christianbundy** [tootsuite/mastodon#13207]
Sorry for the provocative title. :~)

### Expected behaviour

When I select "opt-out of search engine indexing" then no search engine (including other Mastodon instances) should be able to index my toots.

### Actual behaviour

If I understand correctly, this only adds an entry to `robots.txt` asking that search engines don't index my toots. Other Mastodon instances still index my toots without any problems (I think?).

### Steps to reproduce the problem

N/A

### Specifications

mastodon@latest

---

I think it's very important for less-technical users that we explicitly inform them of what privacy guarantees are afforded by this setting and which are not. Friendly search engines that respect `robots.txt` probably won't index toots, but malicious search engines and spiders and other Mastodon instances (!) will still index toots. I think this is important to be very explicit about.

**Comment by Openmedianetwork** [comment]
To be clear you are asking for a rewording and not a code change?

**Comment by christianbundy** [comment]
That's correct, it's important to me that people understand exactly what privacy guarantees they have. Even just a small amount of detail describing something like:

> Some search engines and ActivityPub instances may not respect your request to opt out of indexing. Any search engine or ActivityPub that wants to index your toots can still search through them. Please toot responsibly.

It needs some brevity but I think we need to be up front about how much privacy we can guarantee.

**Comment by ariasuni** [comment]
I thought about the wording of this specific option a while ago and forgot about it.

Here are my reflections about it, mostly from an [old thread I started in French](https://eldritch.cafe/@melunaka/102553670902823515).

- Indexing is technical jargon; it should be very clear what is actually happening
- The option doesn’t tell clearly what data can appear in search results
- The description below the option, “Affects your public profile and status pages”, is a bit unclear and doesn’t tell that only public toots can be found by search engines
- If think opt-out doesn’t convey the idea that any tool can ignore the choice of the user.

Here are a few options:

> [x] Tell search engines that I don’t want my profile and my publicly visible toots to appear in their search results

> [x] Tell search engines to not (save nor) show my profile and publicly visible toots among their/in their/to provide search results

> [x] Tell search engines to not show my profile or my publicly visible toots in their search results
> Search engines browse and save web pages to propose them as search results. Note that they can’t access your followers-only toots or direct messages.

**Comment by skopf** [comment]
I would strongly suggest to completely remove this option as it gives a false sense of privacy.

Recent discussions in the Fediverse around search engines show impressively how this feature is misunderstood by users. This option is only setting the "noindex" meta tag on the profile page, but it does NOT prevent the profile from being indexed by other Mastodon instances.

Another problem is that this option has (almost) no impact on indexing by Google, although Google fully respects this option. Toots and profile information are not just displayed on the users profile page. If a post contains a tag, the post will also appear on the tag page. If a user has at least one follower on a remote instance, the toots will be federated there and be displayed on the federated timeline of the remote instance, from which Google will pick it up. Toots can also be boosted, or replied to, or...
Toots are simply displayed on too many pages to be able to hold up with the promise that they do not end up in Google.

I described this specific problem together with other privacy concerns here:
https://medium.com/@s.kopf/mastodon-is-a-zero-privacy-network-here-is-what-you-need-to-know-86aeed685e41
